### DISCUSSION TITLE
Exception: Undefined property: Livewire\Redirector::$headers error when redirecting from custom exception handler

### DISCUSSION THREAD

**Discussion by telkins** [livewire/livewire#4905] [7. Bugs (archived)]
I get the exception, `Exception: Undefined property: Livewire\Redirector::$headers`, when redirecting from a custom exception handler.

I'm trying to capture a specific exception (query timeout) that can happen in several different places. I can successfully do that by adding the following to Laravel's exception handler class's `register()` method:

```php
    $this->renderable(function (MyInterestingException $e, $request) {
        // ...

        return redirect()->route('the-route-to-be-redirected-to');
    });
```

Versions:

- Laravel 9
- Livewire 2.10.5
- PHP 8.1

Here's a bit of the stack trace:
```
[2022-04-21 11:53:26] local.ERROR: Undefined property: Livewire\Redirector::$headers {"userId":1,"exception":"[object] (ErrorException(code: 0): Undefined property: Livewire\\Redirector::$headers at /var/www/html/vendor/laravel/framework/src/Illuminate/Foundation/Http/Middleware/VerifyCsrfToken.php:194)
[stacktrace]
#0 /var/www/html/vendor/laravel/framework/src/Illuminate/Foundation/Bootstrap/HandleExceptions.php(231): Illuminate\\Foundation\\Bootstrap\\HandleExceptions->handleError()
#1 /var/www/html/vendor/laravel/framework/src/Illuminate/Foundation/Http/Middleware/VerifyCsrfToken.php(194): Illuminate\\Foundation\\Bootstrap\\HandleExceptions->Illuminate\\Foundation\\Bootstrap\\{closure}()
#2 /var/www/html/vendor/laravel/framework/src/Illuminate/Foundation/Http/Middleware/VerifyCsrfToken.php(80): Illuminate\\Foundation\\Http\\Middleware\\VerifyCsrfToken->addCookieToResponse()
#3 /var/www/html/vendor/laravel/framework/src/Illuminate/Support/helpers.php(302): Illuminate\\Foundation\\Http\\Middleware\\VerifyCsrfToken->Illuminate\\Foundation\\Http\\Middleware\\{closure}()
#4 /var/www/html/vendor/laravel/framework/src/Illuminate/Foundation/Http/Middleware/VerifyCsrfToken.php(82): tap()
```

Edit: I forgot to mention that I'm using Livewire in this case via https://filamentphp.com.

Any help/guidance/advice would be appreciated. Thx. 🤓

**Comment by joshhanley** [comment]
@telkins any chance you can share your component class and view and what you did to trigger that error, as well as your custom exeception handler (or better yet a small example repo), so we can investigate?

Thanks!

**Reply by telkins** [reply]
@joshhanley Thx for replying.  I'll try to get something to share as soon as I can. 🤓

**Reply by joshhanley** [reply]
@telkins no worries! Ok great 🙂

**Reply by telkins** [reply]
@joshhanley Just jumping in to say that I haven't forgotten...I've just not yet had the time.  Sorry.  Hope to get to it soon.  It just happened again in a different situation.

**Comment by 4m1n0s** [comment]
Any solution for that?

**Reply by joshhanley** [reply]
@iamine not at this stage. Can you describe the issue you are having and what your context/ scenario is? Thanks!

**Comment by telkins** [comment]
Hey....I should first apologize for not yet sharing the code like I was asked so many months ago.  Things have been quite hectic and, to be honest, I haven't experienced this problem for quite a while.

I'm not sure what to do at this point.  It would take work for me to try to reproduce it at this stage.  If it happens again, though, then I'll try to take the time right then and there to share the component class and exception handler, perhaps via a demo repo as suggested.

**Reply by joshhanley** [reply]
@telkins all good! 🙂 been pretty busy for me too. No worries if you don't manage to reproduce it 🙂

**Comment by bobwurtz** [comment]
@telkins I just ran into this issue as well using Filament. I submitted an [issue](https://github.com/filamentphp/filament/issues/4872). Looks like a Filament problem not a Livewire problem.

**Comment by warmwhisky** [comment]
This issue seems to comes after I have added a route, but only sometimes!? If I clear all caches and cookies I get the error, but when I refresh I get `Route [pos_stations.transfer.choose_station] not defined` however the route is definately defined and was working on office PC. Pulling changes to different PC I get this strange sequence of errors

It seems that something is being cached or broken.

Edit: I am using Sail with WSL on Windows. After running `sail down` then `sail up` it was ok

**Comment by Jastkast** [comment]
In case of anyone still having this issue, it is happening because Livewire is using its own custom Redirector, which apparently has some issues. The error could be fixed by using the original Illuminate\Routing\Redirector 
`return (new Redirector(App::get('url')))->route('elsewhere');`

**Comment by donnyxray** [comment]
Do any of you use Laravel Cashier? That seems to trigger the issue for me. As soon as I have composer remove Laravel Cashier this problem (and a few others) disappears.

**Comment by magic-thomas** [comment]
In my case, it happens in `AuthorizesRequests`.

```php
// Livewire Component. 
    function deletePost()
    {
       return  $this->authorize('delete', [ $this->post, $this->post_cat ] );
}

```

```php
// PostPolicy.php 

public function delete(User $user, Post $post, PostCat $post_cat): bool
    {
        return false;
    }
```

When it get false in Policy, livewire get lost.
